src/ol/source/TileImage.js

```diff
@@ -83,7 +83,8 @@ class TileImage extends UrlTile {
       url: options.url,
       urls: options.urls,
       wrapX: options.wrapX,
-      transition: options.transition
+      transition: options.transition,
+      opt_key: options.opt_key
```

Comment:
I think this property name should just be `key` (we only use the `opt_` prefix for function params).  In addition, the property should be added to the [`@typedef` above](https://github.com/openlayers/openlayers/blob/7d77d05a575ab98391ae9de4038585bd2bfa1bee/src/ol/source/TileImage.js#L17-L55) with a bit of documentation describing what it does.